src/main/java/org/csanchez/jenkins/plugins/kubernetes/AllContainersRunningPodWatcher.java

```diff
@@ -72,7 +84,29 @@ boolean areAllContainersRunning(Pod pod) {
         }
         for (ContainerStatus containerStatus : containerStatuses) {
             if (containerStatus != null) {
-                if (containerStatus.getState().getWaiting() != null) {
+                ContainerStateWaiting waitingState = containerStatus.getState().getWaiting();
+                if (waitingState != null) {
+                    String waitingStateMsg = waitingState.getMessage();
+                    if (waitingStateMsg != null && waitingStateMsg.contains("Back-off pulling image")) {
+                        runListener.error("Unable to pull Docker image \""+containerStatus.getImage()+"\". Check if image name is spelled correctly");
+                        Jenkins jenkins = Jenkins.getInstanceOrNull();
+                        if (jenkins != null) {
+                            Queue q = jenkins.getQueue();
+                            for (Queue.Item item : q.getItems()) {
+                                Label itemLabel = item.getAssignedLabel();
+                                if (itemLabel != null && isCorrespondingLabels(itemLabel.getDisplayName(), pod.getMetadata().getName())) {
```

Comment:
I understand the intent, but this is out of line here. Eventually could be added to exception handling in the launcher https://github.com/jenkinsci/kubernetes-plugin/blob/3225773a140eb590a4d2aaa8281e386812214f97/src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java#L200-L208

Reply:
Are you saying that the KubernetesLauncher should be in charge of canceling the job? We've seen that unless a job is canceled, it'll get put back in the queue and infinitely fail due to that bad pod template.

Reply:
It would be a better place than in `AllContainersRunningPodWatcher`, certainly.